menas/src/main/scala/za/co/absa/enceladus/rest/controllers/HDFSController.scala

```diff
@@ -34,13 +36,26 @@ class HDFSController @Autowired() (fs: FileSystem) {
   private val logger = LoggerFactory.getLogger(this.getClass)
   
   @PostMapping(path = Array("/list"))
-  def getHDFSFolder(@RequestBody path: String): CompletableFuture[HDFSFolder] = {
-    
-    val p = new Path(path)
-    val res = if (!fs.isDirectory(p)) Future { HDFSFolder(p.toUri().getPath, p.getName, None) }
-    else {
+  def getHDFSFolder(@RequestBody pathStr: String): CompletableFuture[ResponseEntity[HDFSFolder]] = {
+    val path = new Path(pathStr)
+
+    Future { fs.exists(path) }.flatMap { exists =>
+      if (exists) {
+        getFolder(path).map(result => ResponseEntity.ok(result))
+      } else {
+        Future.successful(ResponseEntity.notFound().build[HDFSFolder]())
+      }
+    }
+  }
+
+  private def getFolder(path: Path) = {
```

Comment:
What do you think about defining return type of the method ? (Consistency)
Do you have reason why return type of this method is Future ? 
I think you can get rid of them and return only HDFSFolder.
Then method `getHDFSFolder` will look like this: 
`def getHDFSFolder(@RequestBody pathStr: String): CompletableFuture[ResponseEntity[HDFSFolder]] = {
    val path = new Path(pathStr)

    Future {
      if (fs.exists(path)) {
        ResponseEntity.ok(getFolder(path))
      } else {
        ResponseEntity.notFound().build[HDFSFolder]()
      }
    }.flatMap(Future.successful)
  }`